src/main/java/tech/stackable/hbase/OpenPolicyAgentAccessController.java

```diff
@@ -0,0 +1,1137 @@
+    return result;
+  }
+
+  @Override
+  public boolean preCheckAndPutAfterRowLock(
+      final ObserverContext<RegionCoprocessorEnvironment> c,
+      final byte[] row,
+      final byte[] family,
+      final byte[] qualifier,
+      final CompareOperator opp,
+      final ByteArrayComparable comparator,
+      final Put put,
+      final boolean result)
+      throws IOException {
+    User user = getActiveUser(c);
+    TableName tableName = c.getEnvironment().getRegionInfo().getTable();
+    LOG.info("preCheckAndPutAfterRowLock: user [{}] on table [{}] for put [{}]", user, put);
+
+    opaAclChecker.checkPermissionInfo(user, tableName, Action.WRITE);
+    return result;
+  }
+
+  @Override
+  public boolean preCheckAndDelete(
+      final ObserverContext<RegionCoprocessorEnvironment> c,
+      final byte[] row,
+      final byte[] family,
+      final byte[] qualifier,
+      final CompareOperator op,
+      final ByteArrayComparable comparator,
+      final Delete delete,
+      final boolean result)
+      throws IOException {
+    User user = getActiveUser(c);
+    TableName tableName = c.getEnvironment().getRegionInfo().getTable();
+    LOG.info("preCheckAndDelete: user [{}] on table [{}] for delete [{}]", user, delete);
+
+    opaAclChecker.checkPermissionInfo(user, tableName, Action.WRITE);
+    return result;
+  }
+
+  @Override
+  public boolean preCheckAndDeleteAfterRowLock(
+      final ObserverContext<RegionCoprocessorEnvironment> c,
+      final byte[] row,
+      final byte[] family,
+      final byte[] qualifier,
+      final CompareOperator op,
+      final ByteArrayComparable comparator,
+      final Delete delete,
+      final boolean result)
+      throws IOException {
+    User user = getActiveUser(c);
+    TableName tableName = c.getEnvironment().getRegionInfo().getTable();
+    LOG.info(
+        "preCheckAndDeleteAfterRowLock: user [{}] on table [{}] for delete [{}]", user, delete);
+
+    opaAclChecker.checkPermissionInfo(user, tableName, Action.WRITE);
+    return result;
+  }
+
+  @Override
+  public void postListNamespaceDescriptors(
+      ObserverContext<MasterCoprocessorEnvironment> ctx, List<NamespaceDescriptor> descriptors) {
+    // allow for all users
+  }
+
+  @Override
+  public void preTruncateTable(
+      ObserverContext<MasterCoprocessorEnvironment> c, final TableName tableName)
+      throws IOException {
+    User user = getActiveUser(c);
+    LOG.info("preTruncateTable: user [{}] on table [{}]", user, tableName);
+
+    opaAclChecker.checkPermissionInfo(user, tableName, Action.CREATE);
+  }
+
+  @Override
+  public void postTruncateTable(
+      ObserverContext<MasterCoprocessorEnvironment> ctx, final TableName tableName)
+      throws IOException {
+    User user = getActiveUser(ctx);
+    LOG.info("postTruncateTable: user [{}] on table [{}]", user, tableName);
+
+    opaAclChecker.checkPermissionInfo(user, tableName, Action.CREATE);
```

Comment:
I believe this is redundant. Upstream only updates the ACL table because the checking has already been done in the `pre` method.

Reply:
Done